arch/x86/64/source/ArchMemory.cpp

```diff
@@ -1,22 +1,20 @@
 #include "ArchMemory.h"
 #include "ArchInterrupts.h"
 #include "kprintf.h"
-#include "assert.h"
 #include "PageManager.h"
 #include "kstring.h"
 #include "ArchThreads.h"
 #include "Thread.h"
 
-PageMapLevel4Entry kernel_page_map_level_4[PAGE_MAP_LEVEL_4_ENTRIES] __attribute__((aligned(0x1000)));
-PageDirPointerTableEntry kernel_page_directory_pointer_table[2 * PAGE_DIR_POINTER_TABLE_ENTRIES] __attribute__((aligned(0x1000)));
-PageDirEntry kernel_page_directory[2 * PAGE_DIR_ENTRIES] __attribute__((aligned(0x1000)));
-PageTableEntry kernel_page_table[8 * PAGE_TABLE_ENTRIES] __attribute__((aligned(0x1000)));
-
+PageMapLevel4Entry kernel_page_map_level_4[PAGE_MAP_LEVEL_4_ENTRIES] __attribute__((aligned(PAGE_SIZE)));
+PageDirPointerTableEntry kernel_page_directory_pointer_table[2 * PAGE_DIR_POINTER_TABLE_ENTRIES] __attribute__((aligned(PAGE_SIZE)));
+PageDirEntry kernel_page_directory[2 * PAGE_DIR_ENTRIES] __attribute__((aligned(PAGE_SIZE)));
+PageTableEntry kernel_page_table[8 * PAGE_TABLE_ENTRIES] __attribute__((aligned(PAGE_SIZE)));
 
 ArchMemory::ArchMemory()
 {
   page_map_level_4_ = PageManager::instance()->allocPPN();
-  PageMapLevel4Entry* new_pml4 = (PageMapLevel4Entry*) getIdentAddressOfPPN(page_map_level_4_);
+  auto new_pml4 = (PageMapLevel4Entry*) getIdentAddressOfPPN(page_map_level_4_);
```

Comment:
why moving to auto here? being explicit would make it more readable?

Reply:
Repeating yourself for no reason is not necessary and can lead to accidental typos.
It's already clear what the type is from the cast imho.
I agree that the change is not necessary, but it's not a bad change either.